index.js

```diff
@@ -1,14 +1,58 @@
-// Ejercicio 2
-async function writeFile (filePath, data, callback) {
+const fs = require("node:fs/promises");
+const path = require("node:path");
+
+const ERROR_CODES = Object.freeze({ NO_ENTRY: "ENOENT" });
 
+// Ejercicio 2
+async function writeFile(filePath, data, callback) {
+  try {
+    await fs.stat(filePath);
+  } catch (err) {
+    if (err.code === ERROR_CODES.NO_ENTRY) {
+      const filePathDirname = path.dirname(filePath);
+      await fs.mkdir(path.join(__dirname, filePathDirname));
+    }
+  }
+  try {
+    await fs.writeFile(filePath, data);
+    return callback(null);
+  } catch (error) {
+    return callback(error);
+  }
 }
 
-// Ejercicio 3
-async function readFileAndCount (word, callback) {
+const MESSAGE_LIST = Object.freeze({
+  NO_PATH: "No se ha especificado el path del archivo",
+  NO_WORD: "No se ha especificado la palabra a buscar",
+});
 
+// Ejercicio 3
+async function readFileAndCount(word, callback) {
+  const NO_WORD_MATCHES = 0;
+  let proccessError = null;
+  const file = process.argv[2];
+  if (!file) {
+    proccessError = MESSAGE_LIST.NO_PATH;
+    return callback(new Error(proccessError), NO_WORD_MATCHES);
+  }
+  if (!word) {
+    proccessError = MESSAGE_LIST.NO_WORD;
+    return callback(new Error(proccessError), NO_WORD_MATCHES);
+  }
+  let reading;
+  try {
+    reading = await fs.readFile(file, "utf-8");
+  } catch (error) {
+    return callback(proccessError, NO_WORD_MATCHES);
```

Comment:
En este caso, si el archivo no existe debemos pasar 0 sin ningún tipo de error, o es posible que el consumidor de la función piense que ha habido un error aunque en el enunciado dice que devolvemos 0 y ya.

Reply:
Igual también ojo que no se pasa con un `new Error`.

Reply:
Sin embargo una cosita @midudev , en ese caso en la linea 32  donde defino `proccessError` como null, si llega a esa parte del código que mencionas , no le estaría enviando ningún error al consumidor ya que hasta ese momento el valor de `proccessError` se mantiene null por ende, le seguiría enviando tan solo el valor esperado "0" sin ningún error.

Se que parece algo confuso jejej, pero en su momento me pareció una gran idea , crees que este bien hacerlo así o hay alguna mejor forma, quizás definiendo los valores de esta manera un poco más directa sea mejor.
```suggestion
    return callback(null, 0);
```
Es la primera vez que veo node y por lo que estoy aprendiendo, es que es fundamental el control de errores y también como manejamos la respuesta para facilitarle la vida a los consumidores, gracias @midudev por la observación.😊